receiver.go

```diff
@@ -677,13 +703,14 @@ func (r *Receiver) muxHandleFrame(fr frames.FrameBody) error {
 
 	// flow control frame
 	case *frames.PerformFlow:
+		// if the 'drain' flag has been set in the frame sent to the _receiver_ then
+		// we signal whomever is waiting (the service has seen and acknowledged our drain)
+		if fr.Drain && !r.autoSendFlow {
+			r.l.linkCredit = 0 // we have no active credits at this point.
+			r.creditor.EndDrain()
+		}
+
 		if !fr.Echo {
```

Comment:
It seems like the previous check was correct, in that an echo frame might have the drain flag set but that doesn't indicate that _we_ should drain.

Is there some nuance here?

Reply:
I was removing the extra indentation as I believe it's unnecessary (I have some other pending changes in this area when I noticed it).  Only the receiver can set the drain flag, so we should never receive a flow frame from the sender with this and echo set to true.  This is purely cosmetic though (assuming I'm correct), so I can change it back if you prefer.

Reply:
I reverted it.